File: _codigos/entrada_funcional1/calculator.hs
operations = [(\a b -> a + b), (\a b -> a - b), (\a b -> a * b), (\a b -> a / b)]

--testing
operationsNames = ["add", "sub", "mul", "div"]
testOperations a b = [(operations !! i) a b | i <- [0..3]]
--
hazOperaciones a b = zip (testOperations a b) operationsNames

File: _codigos/entrada_funcional1/calculator.cs
IDictionary<string, Func<int, int, int>> calculadoraFuncional = new Dictionary<string, Func<int, int, int>>();
calculadoraFuncional["add"] = (a, b) => a + b;
calculadoraFuncional["sub"] = (a, b) => a - b;
calculadoraFuncional["mul"] = (a, b) => a * b;
calculadoraFuncional["div"] = (a, b) => a / b;

// sum example
int result = calculadoraFuncional["sum"](3, 4);
